.github/workflows/ci.yaml

```diff
@@ -0,0 +1,48 @@
+name: ci
+
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      proto_ref:
+      proto_sha:
+  push:
+    branches:
+      - main
+      - v*
+    # Publish `v1.2.3` tags as releases.
+    tags:
+      - v*
+  # Run tests for PRs
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Info
+        run: echo "Using proto ${{ github.event.inputs.proto_ref }} with sha ${{ github.event.inputs.proto_sha }}"
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Install dependencies
+        run: |
+          # Install buf cli
```

Comment:
What's the advantage of doing all this manually instead of using `mage-loot` like we do for all other languages?
It feels like duplicating logic and diverging from a pattern that seems to serve us well everywhere else.
`mage` makes it easier to manage dependencies because it doesn't require the tool chain to be in `$PATH` which means different projects can use different versions of the tools.

How would I generate the Java bindings locally on my machine from my local pb-authorizer?

Reply:
For the `java-authorizer` we only need the `buf` cli as a dependency that is why I avoided bringing in golang and mage.

Under the hood, mage uses the buf cli. For local development, to generate Java bindings from pb-authorizer you have to 
- in `pb-authorizer` run `mage build`
- in the `java-authorizer` run `buf generate <path to authorizer.bin>`

Reply:
I agree that the `mage generateDev` workflow would not nice to remain the same, that said it does make it a very foreign experience for the main audience being Java developers.

There are `buf` GitHub actions that can mitigate the CI angle, but not the local development one.

See https://buf.build/docs/ci-cd/github-actions

Reply:
In order to keep a good balance between our dev experience and the java ecosystem I did the following:
- added mage with `deps`, `generate`, `generatedev` and `builddev` commands
- made the path to buf executable a property in maven so it is easy to change and set it as default `.ext/gobin/buf-v1.9.0/buf`

This approach has the following advantages:
- we can generate the grpc bindings from mage `mage generate` or from maven `maven package`
- local development is easy because we can use `mage generatedev`
- everything that can be done from mage can also be done from maven except the dependencies download `mage deps`